src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -268,4 +269,22 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
 
         return responses;
     }
+
+    private List<FeedbackResponseAttributes> getSortedExistingResponses(FeedbackQuestionAttributes questionAttributes) {
+        List<FeedbackResponseAttributes> existingResponses = bundle.questionResponseBundle.get(questionAttributes);
+        Map<String, String> emailNamePair = this.bundle.getSortedRecipientList(questionAttributes.getFeedbackQuestionId());
+
+        Comparator<FeedbackResponseAttributes> responsesComparator = (response1, response2) -> {
+            if (emailNamePair.containsKey(response1.recipient) && emailNamePair.containsKey(response2.recipient)) {
+                return emailNamePair.get(response1.recipient).compareTo(emailNamePair.get(response2.recipient));
+            } else {
+                // if the recipients list does not contain the recipient, compare recipients by email
+                // this only could happen due to failures in the test data, should never happen in production
```

Comment:
This doesn't seem right to me.
I would say, lets fix the test case also in this PR, although I am not quite sure how much effort that would require given that other test cases will also be affected